### PR TITLE
Usage of fetchPacks function

### PR DESCRIPTION
**Pull request by MohammedJ18**

### CODE REVIEW

File: src/runtime/composables/useAppManager.ts [hunk 1]
```diff
@@ -5,108 +5,118 @@ import Service from "../models/Service"
...
+      this.apps.push(...this.developmentApps)
+    },
+    async fetchPacks() {
+      // const { $toast } = useNuxtApp()
+      const supabase = useSupabaseClient()
+      let { data: packs, error } = await supabase
+        .from('packs')
+        .select('*,users_packs(id),packs_apps(app_id),packs_services(service_id)')
+      if (error) { console.log('حدث خطأ اثناء تحميل الباقات'); return false }
+      this.packs = packs.map(pack => new Pack(pack))
```
Comment (Gruce) [inline src/runtime/composables/useAppManager.ts:69]: You have used `Pack` model but you didn't imported it. You have to import it.

Comment (Gruce) [reply src/runtime/composables/useAppManager.ts:69]: You have selected `packs_apps(app_id)`. That is good, but you really need to select
```
*, packs_apps (
          id,
          app:app_id (
              *,
              users_apps(id),
              apps_services(
                  id, app_id, title, points, icon, description, users_services(id)
              )
          )
      )
```

And then you have to edit your `Pack Model` as the following comment.

File: src/runtime/models/Pack.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  constructor(args){
+    this.id = args?.id
+    this.title = args?.title
+    this.description = args?.description
+    this.icon = args?.icon
+    this.points = args?.points
+    this.discount = args?.discount
+
+    this.apps = args.apps?.map(app => new App(app))
```
Comment (Gruce) [inline src/runtime/models/Pack.ts:22]:
```
    this.apps = args?.packs_apps?.map(pack_app => new App(pack_app.app))
```

**Review by Gruce — CHANGES_REQUESTED** [review]
Hello @MohammedJ18 ,

I have reviewed your code and it needs some changes. I have provided you enough information and you can check the comments on the code above.

Thank you.

**Comment by MohammedJ18** [comment]
Hi @Gruce,
I completed the changes. You can check and let me know if there are errors or things that need to be added.
Thank you.

**Pull request merged**